app/routes.py

```diff
@@ -1 +1,274 @@
+
+
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_task_incomplete(task_id):
+    
+    task = validate_model(Task, task_id)
+    
+    if task.completed_at:
+        task.completed_at = None
+        
+
+    db.session.commit()
+
+    return {"task": task.to_dict()}
+
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+    try:
+        new_goal = Goal.from_dict_goals(request_body)
+
+    except:
+        abort(make_response({"details": "Invalid data"}, 400))
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    
+    return make_response(jsonify({"goal": new_goal.to_dict_goals()}),201)
+
+@goals_bp.route("", methods=["GET"])
+
+def read_all_goals():
+    title_query = request.args.get("title")
+    sort_filter = request.args.get("sort")
+    
+    if title_query:
+        goals = Goal.query.filter_by(title=title_query)
+    else:
+        goals = Goal.query.all()
+
+    if sort_filter:
+        if sort_filter == "asc":
+            goals = Goal.query.order_by(Goal.title.asc())
+        elif sort_filter == "desc":
+            goals = Goal.query.order_by(Goal.title.desc())
+    else:
+        goals = Goal.query.all()
+    
+    goals_response = []
+    for goal in goals:
+        goals_response.append(goal.to_dict_goals())
+    
+    return jsonify(goals_response)
+
+@goals_bp.route("/<goal_id>", methods=["GET"])
+
+def read_one_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    return {"goal": goal.to_dict_goals()}
+
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    request_body = request.get_json()
+
+    goal.title = request_body["title"]
+    
+
+    db.session.commit()
+
+    return {"goal": goal.to_dict_goals()}
+
+@goals_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    
+    db.session.delete(goal)
+    db.session.commit()
+
+    return make_response({"details":f'Goal {goal.goal_id} "{goal.title}" successfully deleted'})
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"])
+def create_goal_by_id(goal_id):
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json()
+
+    for task in request_body["task_ids"]:
+        task = validate_model(Task, task)
+        goal.tasks.append(task)
+    
+    db.session.commit()
+    return {
+        "id": goal.goal_id,
+        "task_ids": request_body["task_ids"]
+    }
+
+
+@goals_bp.route("/<goal_id>/tasks", methods=["GET"])
+def get_all_tasks_with_id(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    # return {"goal": goal.to_dict_goals_tasks()}
+
+    goals_tasks = {
+        "id": goal.goal_id,
+        "title": goal.title,
+        "tasks": []
+
+    }
+
+    for task in goal.tasks:
+        goals_tasks["tasks"].append(task.to_dict_with_goal
+        ())
+
+    return jsonify(goals_tasks), 200
+
+    
```

Comment:
Well done on this project Phalesa, I didn't have much to comment on and that is a good thing! Keep up the good work! Really looking forward to what you create in the frontend! Please feel free to reach out if you have any questions about the feedback that I left! ✨💫🤭